ipamd/ipamd.go

```diff
@@ -582,44 +579,36 @@ func (c *IPAMContext) increaseIPPool() {
 		return
 	}
 
-	if c.maxENI == c.dataStore.GetENIs() {
-		log.Debugf("Skipping increase IP pool due to max ENI already attached to the instance: %d", c.maxENI)
-		return
-	}
-
-	// Unknown instance type and MAX_ENI is not set
-	if c.maxENI == 0 {
-		log.Errorf("Unknown instance type and MAX_ENI is not set.  Cannot increase IP pool.")
-	}
-
-	if c.dataStore.GetENIs() < c.maxENI {
-		c.tryAllocateENI()
-		c.updateLastNodeIPPoolAction()
-	} else {
-		log.Debugf("Skipping ENI allocation due to max ENI already attached to the instance: %d", c.maxENI)
-	}
-
+	// Try to add more IPs to existing ENIs first.
 	increasedPool, err := c.tryAssignIPs()
 	if err != nil {
 		log.Errorf(err.Error())
 	}
 	if increasedPool {
 		c.updateLastNodeIPPoolAction()
+	} else {
+		// If we did not add an IP, try to add an ENI instead.
+		if c.dataStore.GetENIs() < c.maxENI {
+			c.tryAllocateENI()
+			c.updateLastNodeIPPoolAction()
+		} else {
+			log.Debugf("Skipping ENI allocation due to max ENI already attached to the instance: %d", c.maxENI)
```

Comment:
Trying to read this and it keeps feeling weird. Thoughts on this?

"Skipping ENI allocation as the max ENI limit (%d) was reached"
or 
"Skipping ENI allocation as the instance's max ENI limit (%d) was reached"